script.js

```diff
@@ -1,4 +1,553 @@
+    rank: 6,
+  },
+  {
+    name: "7",
+    suit: "Spades",
+    rank: 7,
+  },
+  {
+    name: "8",
+    suit: "Spades",
+    rank: 8,
+  },
+  {
+    name: "9",
+    suit: "Spades",
+    rank: 9,
+  },
+  {
+    name: "10",
+    suit: "Spades",
+    rank: 10,
+  },
+  {
+    name: "Jack",
+    suit: "Spades",
+    rank: 10,
+  },
+  {
+    name: "Queen",
+    suit: "Spades",
+    rank: 10,
+  },
+  {
+    name: "King",
+    suit: "Spades",
+    rank: 10,
+  },
+];
+
+// Create a function to shuffle a deck of cards.
+function getRandomIndex(max) {
+  return Math.floor(Math.random() * max);
+}
+
+function shuffleCards(cardDeck) {
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    var randomIndex = getRandomIndex(cardDeck.length);
+    var randomCard = cardDeck[randomIndex];
+    var currentCard = cardDeck[currentIndex];
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    currentIndex = currentIndex + 1;
+  }
+  return cardDeck;
+}
+
+var shuffledDeck = shuffleCards(deck);
+console.log(shuffledDeck);
+
+//Define game modes
+var START_GAME = "start game";
+var gameMode = START_GAME;
+var PLAYER_HITS = "player hits";
+var DEALER_HITS = "dealer hits"; // rule: if score after 3rd card < 17, dealer has to hit 4th card.
+var playerHasBusted = false;
+var dealerHasBusted = false;
+
+// List of global variables
+var playerDrawnCards = []; // shuffledDeck.pop(), shuffledDeck.pop()
+console.log(playerDrawnCards);
+var dealerDrawnCards = [];
+console.log(dealerDrawnCards);
+
+// Create a function to calculate score of drawn cards.
+function calcTotalScore(drawnCards) {
+  var numAcesInDrawnCards = 0;
+  var totalScore = 0;
+  for (var i = 0; i < drawnCards.length; i += 1) {
+    var currentCard = drawnCards[i];
+    if (currentCard.name == "Ace") {
+      numAcesInDrawnCards += 1;
+      totalScore += 11;
+    } else {
+      totalScore += currentCard.rank;
+    }
+  }
+  if (totalScore > 21 && numAcesInDrawnCards > 0) {
+    for (
+      var aceCounter = 0;
+      aceCounter < numAcesInDrawnCards;
+      aceCounter += 1
+    ) {
+      totalScore -= 10;
+      if (totalScore <= 21) {
+        break;
+      }
+    }
+  }
+  return totalScore;
+}
+
+// Create a function to check winner
+var finalScore_player = 0;
+var finalScore_dealer = 0;
+var currentScore_player = 0;
+function getWinner(finalScore_player, finalScore_dealer) {
+  var finalScore_player = calcTotalScore(playerDrawnCards);
+  var finalScore_dealer = calcTotalScore(dealerDrawnCards);
+  if (finalScore_player > finalScore_dealer) {
+    return "Player wins!";
+  } else if (finalScore_player < finalScore_dealer) {
+    return "Dealer wins.";
+  } else {
+    return "It's a tie.";
+  }
+}
+
+// Create a function to check Blackjack.
+function checkBlackjack(drawnCards) {
+  if (drawnCards.length == 2 && calcTotalScore(drawnCards) == 21) {
+    var isBlackjack = true;
+    return isBlackjack;
+  } else return (isBlackjack = false);
+}
```

Comment:
The best way to write this function is to return the condition in the if statement. i.e.
`return drawnCards.length == 2 && calcTotalScore(drawnCards) == 21`